Sprint 11

Neenjahw opened the pull request:
На ревью 11 спринт

Neenjahw committed:
sprint_08
1. Сверстан экран LaunchScreen
2. Сверстан экран главного приложения

Neenjahw committed:
no message

Neenjahw committed:
Добавлен градиент для dateLable

Neenjahw committed:
Спринт 9 на ревью

Neenjahw committed:
no message

Neenjahw committed:
Добавил констрейнты ширины и высоты для кнопки поделиться

Neenjahw committed:
Сдаем на ревью спринт 10

Neenjahw committed:
Изменил путь до info.plist

Neenjahw committed:
Добавлен super.viewDidLoad() в WebViewController
Сделан class ImageListCell final

Neenjahw committed:
На ревью 11 спринт

Helvetii951 commented on ImageFeed/Auth/AuthView/AuthViewController.swift:
```diff
@@ -0,0 +1,63 @@
+    private let ShowWebViewSegueIdentifier = "ShowWebView"
+    
+    //MARK: - Override Methods
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == ShowWebViewSegueIdentifier {
+            guard let webViewViewController = segue.destination as? WebViewViewController else {
+                fatalError("Failed for prepare \(ShowWebViewSegueIdentifier)")
+            }
+            webViewViewController.delegate = self
```
`fatalError` лучше никогда не использовать. Может привести к очень неприятным падениям в будущем. Рекомендую использовать [assertionFailure](https://developer.apple.com/documentation/swift/assertionfailure(_:file:line:)). 
Часто в таких случаях пишут ошибки в логи или сервисы статистики и событий.

Helvetii951 commented on ImageFeed/SingleImage/SingleImageViewController.swift:
```diff
@@ -0,0 +1,92 @@
+    }
+    
+    //MARK: - IBOutlets
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet private var imageView: UIImageView!
+    @IBOutlet private var buttonShare: UIButton!
```
Свойство `scrollView` можно сделать приватным. Рекомендую всегда писать функции и свойства с минимальной областью видимости, а затем менять ее, если нужно.

Helvetii951 commented on ImageFeed/Auth/WebView/WebViewController.swift:
```diff
@@ -0,0 +1,108 @@
+//MARK: - class WebViewViewController
+final class WebViewViewController: UIViewController {
+    
+    //MARK: - Private Constants
+    enum WebViewConstants {
+        static let unsplashAuthorizeURLString = "https://unsplash.com/oauth/authorize"
+    }
```
Не такая уж эта костанта и приватная :) В проекте есть файл констант, я бы рекомендовал вынести ее туда. Тогда тебе будет легче найти и поменять, если урл поменяется.

Helvetii951 commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,101 @@
+    //MARK: - Override Mehods
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == ShowSingleImageSegueIdentifier {
+            let viewController = segue.destination as! SingleImageViewController
+            let indexPath = sender as! IndexPath
+            let imageName = photosName[indexPath.row]
+            let image = UIImage(named: "\(imageName)_full_size") ?? UIImage(named: imageName)
+            viewController.image = image
```
Очень рекомендую избегать любых force операций. Они могут привести к неожиданным падениям в продакшене. Здесь лучше использовать `as?`, а затем проверять с помощью guard или if-let.

Helvetii951 commented on ImageFeed/Profile/ProfileView/ProfileViewController.swift:
```diff
@@ -0,0 +1,139 @@
+            let profileImageURL = ProfileImageService.shared.avatarURL,
+            let url = URL(string: profileImageURL)
+        else { return }
+        let processor = RoundCornerImageProcessor(cornerRadius: 60)
+        avatarImageView.kf.indicatorType = .activity
```
RoundCornerImageProcessor из Kingfisher скругляет именно ту картинку, которую скачал. Т.е., если он скачал большую картинку(например, 2000х2000), то он скруглит у нее только 60 поинтов(такое число указано в коде проекта), что вряд ли будет заметно. Всегда учитывай это при использовании RoundCornerImageProcessor.

Helvetii951 commented on ImageFeed/Splash/SplashViewController.swift:
```diff
@@ -0,0 +1,112 @@
+    //MARK: - Private Properties
+    private let showAuthenticationScreenSegueIdentifier = "showAutentificationScreen"
+    
+    private let splashImageLogo: UIImageView = {
+        let splashImage = UIImage(named: "splash_screen_logo")
+        let view = UIImageView(image: splashImage)
+        return view
+    }()
```
Рекомендую всегда использовать [ленивые свойства ](https://www.avanderlee.com/swift/lazy-var-property/)для объявления подобных вьюх.  В нашем случае это какого-то буста не даст. Но, если какая-то вьюха/свойство не будет использоваться, можно сэкономить память.

Neenjahw committed:
1. Изменил fatalError на assertionFailure в AuthViewController
2. Сделал свойство scrollView приватным в SingleImageViewController
3. Избавился от force unwrap в ImageListViewController
4. Перенес свойство static let unsplashAuthorizeURLString в файл Constants
